services/load/service.go

```diff
@@ -0,0 +1,666 @@
+package load
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/influxdata/kapacitor/client/v1"
+	kexpvar "github.com/influxdata/kapacitor/expvar"
+	"github.com/influxdata/kapacitor/server/vars"
+	"github.com/influxdata/kapacitor/services/storage"
+	"github.com/pkg/errors"
+)
+
+var defaultURL = "http://localhost:9092"
+
+const (
+	statErrorCount = "errors"
+)
+
+const (
+	// Public name of overrides store
+	loadAPIName = "load"
+	// The storage namespace for all configuration override data.
+	loadNamespace = "load_items"
+)
+
+type Diagnostic interface {
+	Debug(msg string)
+	Error(msg string, err error)
+	Loading(thing string, file string)
+}
+
+type Service struct {
+	mu     sync.Mutex
+	config Config
+
+	cli        *client.Client
+	statsKey   string
+	statMap    *kexpvar.Map
+	errorCount *kexpvar.Int
+
+	items ItemsDAO
+
+	tasks     map[string]bool
+	templates map[string]bool
+	handlers  map[string]bool
+
+	StorageService interface {
+		Store(namespace string) storage.Interface
+		Register(name string, store storage.StoreActioner)
+	}
+
+	diag Diagnostic
+}
+
+func NewService(c Config, h http.Handler, d Diagnostic) (*Service, error) {
+	cfg := client.Config{
+		URL: defaultURL,
```

Comment:
Can we add a unique user agent here so its clear the consumer of the API is this internal client from Load?

Reply:
fixed